src/main/java/org/opentripplanner/middleware/bugsnag/BugsnagDispatcher.java

```diff
@@ -0,0 +1,213 @@
+package org.opentripplanner.middleware.bugsnag;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.opentripplanner.middleware.bugsnag.response.Organization;
+import org.opentripplanner.middleware.bugsnag.response.Project;
+import org.opentripplanner.middleware.models.BugsnagEvent;
+import org.opentripplanner.middleware.models.BugsnagEventRequest;
+import org.opentripplanner.middleware.utils.HttpUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsInt;
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsText;
+
+/**
+ * Class to provide Bugsnag information by constructing required API calls to Bugsnag
+ */
+public class BugsnagDispatcher {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BugsnagDispatcher.class);
+
+    private static final String BUGSNAG_API_URL = "https://api.bugsnag.com";
+    private static final String BUGSNAG_API_KEY = getConfigPropertyAsText("BUGSNAG_API_KEY");
+    private static final String BUGSNAG_ORGANIZATION = getConfigPropertyAsText("BUGSNAG_ORGANIZATION");
+    private static final int BUGSNAG_REPORTING_WINDOW_IN_DAYS
+        = getConfigPropertyAsInt("BUGSNAG_REPORTING_WINDOW_IN_DAYS", 14);
+
+    private static Organization ORGANIZATION = null;
+    private static HashMap<String, Project> PROJECTS = new HashMap<>();
```

Comment:
It would be good for a lot of these variables to have JavaDoc. With this `PROJECTS` variable in particular, it would be good to note that it is only ever set during startup.

Reply:
Updated.